### ISSUE TITLE
Hero block accepts tokens but doesn't say so

### ISSUE THREAD

**Issue by BWPanda** [backdrop/backdrop-issues#4311]
The Hero block [accepts tokens](https://github.com/backdrop/backdrop/blob/1.x/core/modules/layout/includes/block.hero.inc#L38) in its Block Content field, but nowhere does it mention that, nor is there any links to a token browser so you can easily insert tokens. We should fix that :smiley: 

I'm not sure if the same is true for other blocks or not, just noticed it here because of a [discussion in Gitter](https://gitter.im/backdrop/backdrop-issues?at=5e46ef90d56ddb68a4a6311e).

**Comment by klonos** [comment]
It turns out that all custom blocks support tokens as well 😉 

PS: (separate issue for this, because it'd be a new feature instead of a task) should we make it so that the block titles support tokens too? ...during the UX study @stpaultim did recently, the user attempted to change the hero block title when asked to change the site name. So, I though that if the hero title was set to `Welcome to [site:name]`, it'd change automatically when the site name was updated:

![Screen Shot 2020-02-18 at 4 33 17 pm](https://user-images.githubusercontent.com/2423362/74706988-96e8b480-526c-11ea-9c94-3bddce5bcc0a.png)

**Comment by BWPanda** [comment]
Here's a PR: https://github.com/backdrop/backdrop/pull/4015

However there's an issue:

![Peek 2022-03-24 14-12](https://user-images.githubusercontent.com/2385329/159834702-92db9d34-1b8e-48b9-a52b-79679a9b11cb.gif)

It seems that because of the modal window, it doesn't recognise that you've clicked in the field... Not sure how to fix that. Marking as 'needs work' for now.

**Comment by docwilmot** [comment]
COuld we implement this as a CKEditor plugin with its own icon, instead?

**Comment by BWPanda** [comment]
@docwilmot That assumes CKEditor is enabled and usable, which may not always be the case...

**Comment by klonos** [comment]
> However there's an issue:

@BWPanda I believe that this issue here is blocked on #5973.

**Comment by klonos** [comment]
The PR here doesn't need work - it will in fact work fine once we solve #5973.

**Comment by klonos** [comment]
> ...it will in fact work fine once we solve #5973.

...well, not exactly 🤔 ...the insertion of the token will work, but I don't think that CKEditor fields support substitution of tokens. We might be entering https://github.com/backdrop-contrib/token_filter area here 🤔

**Comment by argiepiano** [comment]
Something to keep in mind:  if you enable the `token_filter` module, and then enable the token filtering functionality for the "Filtered HTML" text format, you'll end up with two token tree links, one provided by `token_filter` and another by the PR here.

This image is what the Hero block editing modal looks like **without** the PR here - only with `token_filter` enabled, and after expanding "Formatting options".  

<img width="594" alt="Screen Shot 2023-02-11 at 9 20 39 PM" src="https://user-images.githubusercontent.com/9938978/218293028-41f5c7cf-1aa5-4f8c-8d79-961049c74cf6.png">

As a side note: notice that the token tree that you get when you click the "Browse tokens" link as provided by Token Filter doesn't actually allow you to insert tokens by clicking. This behavior is true not just for the Hero block, but for any textfield that uses Token Filter to filter tokens. This is probably so because of the difficulties of inserting tokens in ckeditor fields. So, if you still want to add a link to the token tree, you may opt to "turn off" the insert capability, since it doesn't quite work anyway.  You do this like so: 
```php
 theme('token_tree_link', array(
      'click_insert' => FALSE,
));

**Comment by argiepiano** [comment]
>  I don't think that ALL CKEditor fields support substitution of tokens 

Right. Text fields attached to entities do not provide token replacement by default, unless you enable `token_filter`. Blocks are a special case, where several of them will provide global token replacement without the need to enable `token_filter`. For example, custom blocks...

**Comment by BWPanda** [comment]
Happy for someone else to work on this if/when it's unblocked.

**Comment by herbdool** [comment]
The PR https://github.com/backdrop/backdrop/pull/4015 only addresses custom blocks created while in the layout UI. If using the standalone page `admin/structure/block/add` the token link doesn't exist.

I also tried out token_filter but didn't see the link included in the latest release. There are a couple small tweaks that are newer.
